rust/src/writer/utils.rs

```diff
@@ -235,3 +282,113 @@ impl Write for ShareableBuffer {
+            (
+                Arc::new(Int64Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(UInt8Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(UInt16Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(UInt32Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(UInt64Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(StringArray::from(vec![
+                    Some(String::from("1")),
+                    Some(String::from("2")),
+                ])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(BooleanArray::from(vec![Some(true), Some(false)])),
+                Some(String::from("true")),
+            ),
+            (
+                Arc::new(Date32Array::from(vec![Some(1), Some(2)])),
+                Some(String::from("1970-01-02")),
+            ),
+            (
+                Arc::new(Date64Array::from(vec![Some(86400000), Some(2)])),
+                Some(String::from("1970-01-02")),
+            ),
+            (
+                Arc::new(TimestampSecondArray::from(vec![Some(1), Some(2)])),
+                Some(String::from("1970-01-01 00:00:01")),
+            ),
+            (
+                Arc::new(TimestampMillisecondArray::from(vec![Some(1000), Some(2)])),
+                Some(String::from("1970-01-01 00:00:01")),
+            ),
+            (
+                Arc::new(TimestampMicrosecondArray::from(vec![
+                    Some(1000000),
+                    Some(2),
+                ])),
+                Some(String::from("1970-01-01 00:00:01")),
+            ),
+            (
+                Arc::new(TimestampNanosecondArray::from(vec![
+                    Some(1000000000),
+                    Some(2),
+                ])),
+                Some(String::from("1970-01-01 00:00:01")),
+            ),
+            (
+                Arc::new(BinaryArray::from_vec(vec![b"1", b"2"])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(BinaryArray::from_vec(vec![b"\x00\\", b"2"])),
+                Some(String::from("\\x00\\\\")),
+            ),
+            (
+                Arc::new(LargeBinaryArray::from_vec(vec![b"1", b"2"])),
+                Some(String::from("1")),
+            ),
+            (
+                Arc::new(LargeBinaryArray::from_vec(vec![b"\x00\\", b"2"])),
+                Some(String::from("\\x00\\\\")),
+            ),
+        ];
+        for (vals, result) in reference_pairs {
+            assert_eq!(stringified_partition_value(&vals).unwrap(), result)
```

Comment:
You can use `as_deref()` to turn `Option<String>` into `Option<&str>`:
```suggestion
            assert_eq!(
                stringified_partition_value(&vals).unwrap().as_deref(),
                result
            )
```